lib/templatesimplifier.cpp

```diff
@@ -320,6 +329,38 @@ unsigned int TemplateSimplifier::templateParameters(const Token *tok)
     return 0;
 }
 
+// Same as Token::eraseTokens() but tries to fix up lists with pointers to the deleted tokens.
+void TemplateSimplifier::eraseTokens(Token *begin, const Token *end)
+{
+    if (!begin || begin == end)
+        return;
+
+    while (begin->next() && begin->next() != end) {
+        // check if token has a pointer to it
+        if (begin->next()->hasPointer()) {
+            // check if the token is in a list
+            const std::list<TokenAndName>::iterator it2 = std::find_if(mTemplateInstantiations.begin(),
+                    mTemplateInstantiations.end(),
+                    FindToken(begin->next()));
+            if (it2 != mTemplateInstantiations.end()) {
+                // remove the pointer flag and prevent the deleted pointer from being used
+                it2->token->hasPointer(false);
+                it2->token = nullptr;
+            }
+        }
+        begin->deleteNext();
+    }
+}
+
+// The token following tok may have a pointer to it so don't invalidate it.
```

Comment:
I'd write in the comment that tok is invalidated.

Hmm.. it is different behaviour to `Token::deleteThis` so I thought about changing the function name. Maybe `deleteToken(tok)`.

Reply:
Fixed. Revised comment moved to header file.